### PR TITLE
FlexCAN driver for i.MX RT1060 MCU

### PR DESCRIPTION
**Pull request by michallenc**
## Summary
Adds FlexCAN driver for i.MX RT1060 MCU and the support for imxrt1060-evk board.
## Testing
Verified on imxrt1060-evk board, tested againts [MicroZed APO board](https://cw.fel.cvut.cz/b192/_media/courses/b35apo/en/semestral/mz_apo-datasheet-en.pdf)

### CODE REVIEW

**Comment by michallenc** [comment]
Any idea about arch/arm/src/imxrt/hardware/rt106x/imxrt106x_ccm.h warnings? As far as I can see it is the same coding style as it is the mainline. Thanks.

**Comment by davids5** [comment]
Hi @michallenc 

Add a blank line above the line the line that failed OR moved the comment to the right to align with with the previous command's alignment.

**Comment by Ouss4** [comment]
The build is failing becase imxrt1060-evk/canfd defconfig's is using old configurations.
Please refresh it by running: `./tools/refresh.sh --silent imxrt1060-evk:canfd`

**Comment by michallenc** [comment]
Hi @davids5 and @Ouss4 

Thank you for your advices and help.

File: boards/arm/imxrt/imxrt1060-evk/src/imxrt_flexcan.c [hunk 1]
```diff
@@ -0,0 +1,96 @@
+/****************************************************************************
+ * boards/arm/imxrt/imxrt1060-evk/src/imxrt_flexcan.c
+ *
+ *
+ * Redistribution and use in source and binary forms, with or without
```
Comment (xiaoxiang781216) [inline boards/arm/imxrt/imxrt1060-evk/src/imxrt_flexcan.c:5]: Please change the copright to apache.

Comment (michallenc) [reply boards/arm/imxrt/imxrt1060-evk/src/imxrt_flexcan.c:5]: @xiaoxiang781216 Sure. Just by adding
 *   Copyright (C) 2020 Gregory Nutt. All rights reserved.
 *   Author: Michal Lenc <lencmich@fel.cvut.cz>
 *
 * Based on: arch/arm/src/s32k1xx/s32k1xx_flexcan.h

or is there a need to change the whole Apache license as in [Coding Standard](https://cwiki.apache.org/confluence/display/NUTTX/Coding+Standard#cfilestructure)? Thanks.

Comment (acassis) [reply boards/arm/imxrt/imxrt1060-evk/src/imxrt_flexcan.c:5]: Yes, it needs to be changed to Apache license

Comment (michallenc) [reply boards/arm/imxrt/imxrt1060-evk/src/imxrt_flexcan.c]: @acassis Thanks, it´s done.

File: arch/arm/src/imxrt/imxrt_flexcan.h [hunk 2]
```diff
@@ -0,0 +1,125 @@
+/************************************************************************************
+ * arch/arm/src/imxrt/imxrt_flexcan.h
+ *
+ *   Copyright (C) 2020 Gregory Nutt. All rights reserved.
```
Comment (xiaoxiang781216) [inline arch/arm/src/imxrt/imxrt_flexcan.h:4]: need change too

Comment (michallenc) [reply arch/arm/src/imxrt/imxrt_flexcan.h:4]: @xiaoxiang781216 This one is based on s32k1xx_flexcan.h which is under the other (old) licence, so the code should retain the copyright. Or do I understand it wrong?

Comment (patacongo) [reply arch/arm/src/imxrt/imxrt_flexcan.h:4]: > 
> 
> @xiaoxiang781216 This one is based on s32k1xx_flexcan.h which is under the other (old) licence, so the code should retain the copyright. Or do I understand it wrong?

Yes, but I am the author/coyright holder on that file.  I have submitte both and ICLA and an SGA granting all NuttX software to the ASF.  The license has not yet been changed on s32k1xx_flexcan.h but it should be.

I also grant you the right to change the header to Apache 2.0.  No new files may come into the repository with the old BSD license.  So if the header is not changed, the the PR will have to be declined.

File: arch/arm/src/imxrt/hardware/imxrt_flexcan.h [hunk 3]
```diff
@@ -0,0 +1,481 @@
+/****************************************************************************************************
+ * arch/arm/src/imxrt/hardware/imxrt_flexcan.h
+ *
+ *   Copyright (C) 2020 Gregory Nutt. All rights reserved.
+ *   Author: Michal Lenc <lencmich@fel.cvut.cz>
+ *
+ * Based on: arch/arm/src/s32k1xx/hardware/s32k1xx_flexcan.h
```
Comment (xiaoxiang781216) [inline arch/arm/src/imxrt/hardware/imxrt_flexcan.h:7]: Another one.

File: arch/arm/src/imxrt/imxrt_flexcan.h (hunk 2 above)
Comment (michallenc) [reply arch/arm/src/imxrt/imxrt_flexcan.h:4]: @patacongo Thank you for clarifying this. I´ll make the changes.

**Pull request merged**